### ISSUE TITLE
Cache highlight overrides

### ISSUE THREAD

**Issue by fecet** [catppuccin/nvim#664]
If I set highlight overrides in config, say
```lua
highlight_overrides = {
                all = function(cp)
                    return {
                        PmenuSel = { bg = cp.green, fg = cp.base },
                    }
                end,
            }
```
staruptime will significantly slow down by catppuccin by 5-10ms depends on hardware
```
❯ timevim # with override
Total Average: 34.439000 msec
Total Max:     50.645000 msec
Total Min:     25.422000 msec
❯ timevim # without override
Total Average: 29.280420 msec
Total Max:     41.914000 msec
Total Min:     25.985000 mse
```
using `vim-startuptime --vimpath=nvim -count 100 -warmup 5 | grep Total` with
https://github.com/rhysd/vim-startuptime

**Comment by nekowinston** [comment]
AFAIK it should already get cached: https://github.com/catppuccin/nvim/blob/main/lua/catppuccin/lib/compiler.lua / https://github.com/catppuccin/nvim/blob/main/lua/catppuccin/lib/hashing.lua

On my macOS, the compiled cache (including my custom mappings) is stored in `~/.cache/nvim/catppuccin/<flavor name>`

**Comment by fecet** [comment]
Thanks for your quick response. Is this delay expected?

**Comment by EdenEast** [comment]
Catppuccin hashes your user configuration as well as the git repo state. It then only re-compile the theme if the cached hash differs.

You can see how the hash is calculated here https://github.com/catppuccin/nvim/blob/9703f227bfab20d04bcee62d2f08f1795723b4ae/lua/catppuccin/init.lua#L169-L172

If nothing changes then it will execute the pre-compiled lua function stored in its cache.

TLDR: first load after a configuration change or plugin update there will be a re-compile of the cache. Subsequent loads are cached and no delay is expected.

**Comment by nullchilly** [comment]
Thanks!

It should work correctly now

## Before

```lua
vim-startuptime --vimpath=nvim -count 100 -warmup 5 | grep Total
Total Average: 40.001640 msec
Total Max:     59.516000 msec
Total Min:     27.682000 msec
```

## After

```lua
vim-startuptime --vimpath=nvim -count 100 -warmup 5 | grep Total
Total Average: 29.774260 msec
Total Max:     33.727000 msec
Total Min:     26.942000 msec
```
